tests/functional/behave_features/common/utils/owner_file_submission.py

```diff
@@ -0,0 +1,351 @@
+        self.branch_base_id = f"{self.head_sha}-{self.uuid}"
+
+        logging.info(f"base branch identifer: {self.unique_branch()}")
+        # Create the base branch for this workflow run.
+        self.repo_manager.checkout_branch(self.unique_branch())
+        logging.debug(
+            f"Active branch name: {self.repo_manager.repo.active_branch.name}"
+        )
+
+        # Create the branch on the remote if it doesn't exist.
+        r = github.github_api("get", f"repos/{test_repo}/branches", bot_token)
+        branches = json.loads(r.text)
+        branch_names = [branch["name"] for branch in branches]
+        logging.debug(f"Remote test repo branch names : {branch_names}")
+        if self.unique_branch() not in branch_names:
+            logging.info(
+                f"{test_repo}:{self.unique_branch()} does not exists, creating with local branch"
+            )
+            self.repo_manager.push_branch(test_repo, self.unique_branch())
+
+        # Create base branch and pr branch.
+        base_branch, pr_branch = self.workflow_branches(self.test_name)
+        logging.debug(f"base branch is set to: {base_branch}")
+        logging.debug(f"pr branch is set to: {pr_branch}")
+
+        # Set "secrets"
+        #
+        # NOTE(komish): This is a carry-over from the chart certification CI.
+        # Not all items are secrets, but it's unclear why these are designated
+        # as such. Prioritizing consistency for now.
+        self.secrets.test_repo = test_repo
+        self.secrets.bot_name = bot_name
+        self.secrets.bot_token = bot_token
+        self.secrets.base_branch = base_branch
+        self.secrets.pr_branch = pr_branch
+        self.secrets.index_file = "index.yaml"
+
+        self.set_git_username_email(bot_name, settings.GITHUB_ACTIONS_BOT_EMAIL)
+        self.temp_dir = self.repo_manager.add_worktree()
+
+    def unique_branch(self) -> str:
+        """Returns this instance's unique branch name."""
+        return f"e2e-owners-{self.branch_base_id}"
+
+    def workflow_branches(self, test_name: str) -> (str, str):
+        """Returns the base_branch and pr_branch names for a given test_name."""
+        pretty_test_name = self.test_name.strip().lower().replace(" ", "-")
+        base_branch = (
+            f"{self.unique_branch()}-{pretty_test_name}"
+            if pretty_test_name
+            else f"{self.unique_branch()}-test"
+        )
+        return base_branch, f"{base_branch}-pr-branch"
+
+    def cleanup(self):
+        """Cleans up all artifacts that are created locally and remotely."""
+        logging.debug("RedHatOwnersFileSubmissionE2ETest --> cleanup called!")
+        if self.repo_manager:
+            try:
+                self.repo_manager.cleanup()
+            except Exception as e:
+                logging.error(
+                    f"failed to execute cleanup of the repo_manager with error: {e}"
+                )
```

Comment:
So, I considered this case - 

Technically, `self.repo_manager.cleanup()` doesn't throw an exception... that is, unless the hard-coded exceptions we're catching (and reducing down to a log line) don't catch all exceptions.

To that end, I'd like to leave this as is - at least to prevent a stacktrace where it's not necessary. Cleanup failures aren't the end of the world, and shouldn't necessarily fail the CI workflow.

Reply:
> Cleanup failures aren't the end of the world, and shouldn't necessarily fail the CI workflow.

Ok